### ISSUE TITLE
Better guidance for plural translations

### ISSUE THREAD

**Issue by dg0yt** [nijel/weblate#1271]
### Actual behaviour
The source part of the Translate box display the same string in both fields ('One', 'Other'). This is confusing: Uninitiated translators are tempted to translate this text literally, thus missing the point of having plural translations.

![plural-translation](https://cloud.githubusercontent.com/assets/13567791/19957547/f1aa0998-a197-11e6-8d77-1358ca8409a3.png)

### Expected behaviour
- [ ] Don't display the the same source string twice. Possible solutions are:
  - Display the string only once.
  - Use the plural translation which exists in the repository. This requires that the language of the source strings is known. Sometimes there will still be an inconsistency in the plural usage (e.g. en vs. cs), so this might be overkill.
- ~~Provide a check which catches the case that the translations are equal for all singular/plural forms.~~ (Moved to #1273)
- [ ] Add the n-term to each translation headline, e.g. "One (n == 1)", "Other (n != 1)", instead of displaying the formular below all translations.

### Server configuration
hosted

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/38912325-better-guidance-for-plural-translations?utm_campaign=plugin&utm_content=tracker%2F253393&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F253393&utm_medium=issues&utm_source=github).

**Comment by nijel** [comment]
This is only problem for Qt TS files as it's only format which doesn't store separately singular and plural for source language. Weblate or translate-toolkit probably could interpolate the `(s)` to provide expected result.

As for the check, can you please open separate issue for this?

And with the n-term, it's not that easy to do that, that's why there is plural equation printed. Check [plural equation](http://docs.translatehouse.org/projects/localization-guide/en/latest/l10n/pluralforms.html) for Arabic and tell me how this should be interpolated :-).

PS: Generally please try to open one issue per one feature, it makes way easier to track things once one of them is implemented and others not.

**Comment by dg0yt** [comment]
I tried to split the n-terms from `plurals.txt` with `sed`. It doesn't look that bad. There are some unpaired parentheses left, which could be handled more easily by modifying the input file. E.g. Arabic:

~~~
   ar,    Arabic [#f1]_,         nplurals=6; plural=(n==0 ? 0 : n==1 ? 1 : n==2 ? 2 : n%100>=3 && n%100<=10 ? 3 : n%100>=11 ? 4 : 5);
     0 : (n==0 
     1 : n==1 
     2 : n==2 
     3 : n%100>=3 && n%100<=10 
     4 : n%100>=11 
     5) : OTHER n
~~~

[test.sed.txt](https://github.com/nijel/weblate/files/573129/test.sed.txt)
[out.txt](https://github.com/nijel/weblate/files/573128/out.txt)

**Comment by nijel** [comment]
It is really better to split equation like this than listing it on it's own? Maybe Arabic was not best example here (it has many plurals, but pretty simple conditions), for Bosnian, the second plural form formula is `n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20)`.

**Comment by dg0yt** [comment]
Can you suggest a particular translation (string) to look at in the web interface?

The proposal was to have the particular term next to each of the quite generic label. This is much better than having to understand how the complete formular selects from the different translations.
Reviewing whitespace and parens could improve readability in some cases.

If the concern is that this is still too complex for translators or too much clutter, the next approach could be to list a few matching figures next to each label, e.g. for hezky cesky
**One** 1
**Few** 2, 3, 4
**Other** 0, 5, 6...
The lists of figures could be generated by enumeration from zero to some n (like finding prime numbers with the Sieve of Eratosthenes). Knowing the bounds from the plural rules, you could even skip large parts from the range.

Both the terms and or the list could be pregenerated, and the example figures could even be verified.

**Comment by nijel** [comment]
Maybe having number examples would be better than showing part of the equation (still left to figure out with formats which do not store equation at all).

Example for Bosnian:

https://hosted.weblate.org/translate/phpmyadmin/master/bs/?checksum=1118d956618c4f063e67d8e69bb1ce73

**Comment by dg0yt** [comment]
Much better now, great. Ten numbers seems a little bit long to me, but I hope you receive more feedback for this change.

**Comment by nijel** [comment]
I was not sure about how much numbers to show, 10 seemed to me like a good option to be able to present more complex patterns where the first set is quote long, but continues later, for example:

* Arabic: Few (e.g. 3, 4, 5, 6, 7, 8, 9, 10, 103, 104)
* Lithuanian: Few (e.g. 2, 3, 4, 5, 6, 7, 8, 9, 22, 23)
